src/config.rs

```diff
@@ -3042,24 +3042,6 @@ impl TiKvConfig {
     }
 
     pub fn check_critical_cfg_with(&self, last_cfg: &Self) -> Result<(), String> {
-        if last_cfg.rocksdb.wal_dir != self.rocksdb.wal_dir {
```

Comment:
you should find the reason, the check looks reasonable

Reply:
I didn't remove any check.

Reply:
Verified in https://github.com/tikv/tikv/issues/12238.